### DISCUSSION TITLE
OSF-file plugin for dynamic data (e.g. roadworks and detours)?

### DISCUSSION THREAD

**Discussion by Madeo19** [osmandapp/OsmAnd#19412] [Q&A] [answered]
Hi there,
i would like to create an OsmAnd plugin which displays dynamic data like roadworks and detours. Can this be realized with the OBF-file approach? In general, I would like to distribute my plugin via Play Store and App Store similar to the OsmAnd Ski Map Plugin (https://play.google.com/store/apps/details?id=net.osmand.skimapsPlugin&hl=gsw&gl=US). Is this possible with an OBF-file? Beside OBF-files, ChatGPT told me to implement the subclass the OsmandPlugin class and to register the plugin via the AndroidManifest.xml as a service via `<action android:name="net.osmand.plugin.PluginInterface" />` however, PluginInterface does not seem to exist anymore.

**Comment by mariush444** [comment]
You can define (in osf) your own map for download menu.
           "downloadurl": "https:\/\/domane.com\/osmand\/my-map.zip",

The only question is how to define auto refreshing period.
For this moment, you can refresh manualy only.

**Comment by vshcherb** [comment]
Yes it's possible to keep pushing the map update via AIDL interface. And keep plugin running. You need to create OBF file on server side cause otherwise it won't be working. 

We also have code for direction points which auto assign tags to the roads but unfortunately we didn't finish it with API I think

**Reply by vshcherb** [reply]
RoutingConfiguration.setDirectionPoints

**Comment by Madeo19** [comment]
Thanks for your help! I will try to implement and possibly come back with questions concerning certain details.

**Comment by Madeo19** [comment]
Sorry, I was completely wrong! Of course, I meant the **OSF** file format, NOT the OBF file format. I am going to change the topic title.

**Comment by Madeo19** [comment]
In https://osmand.net/docs/user/plugins/custom I read "This package is a collection of **online** and offline map sources of various types." However, the description of the example contains only offline map sources. So, how could **online** map sources defined/included in the plugin?

**Comment by mariush444** [comment, marked as answer]
With osf (plugin file)
you can define access to raster online data
https://osmand.net/docs/user/map/raster-maps/
if you have server that supports tiles like wms wmts and similar.
You can define access to vector obf maps like every country file in osmand.
The only question is how often you can recreate the file and how often user have to/want to update it.
